flake.nix

```diff
@@ -58,6 +58,45 @@
             hardeningDisable = [ "fortify" ];
           };
           treefmtEval = treefmt-nix.lib.evalModule pkgs ./treefmt.nix;
+          mkDevShell =
+            { cmd, editor }:
+            pkgs.mkShell {
+              nativeBuildInputs = [
+                nixd
+                pkgs.nixfmt-rfc-style
+                pkgs.git
+                editor
+              ];
+              inputsFrom = [ config.flake-root.devShell ];
+              shellHook = ''
+                echo -e "\n\033[1;31mDuring the first time nixd launches, the flake inputs will be fetched from the internet, this is rather slow.\033[0m"
```

Comment:
I have some aversion to these shell scripts; it would be better if the logic could be rewritten in Python.